macros/contexts/contextNondecimalBase.pl

```diff
@@ -0,0 +1,273 @@
+expressions), or lists of hexadecimal numbers.  The former allows
+expression involving numeric operations and bitwise operations.
+
+Once one of these contexts is selected, all the nummbers parsed by
+MathObjects will be considered to be in hexadecimal, so
+
+        $n = Compute('10');
+
+produces the hexadecimal number 10 (decimal 16).  You could also obtain
+a hex MathObject using
+
+        $n = Hex(0x10);
+
+Once you have such a value, use
+
+        ANS($n->cmp)
+
+to get an answer checker for the number.  You can also perform numeric or
+bitwise operations on the value, as in
+
+        $m = $n + 0xE3;
+        $N = $n << 2;     # shifts n to the left 2 (binary) places,
+                          #   $N = Hex(0x40) when $n = Hex(0x10)
+
+=cut
+
+sub _contextNondecimalBase_init {
+	context::NondecimalBase::Init();
+	sub setBase     { context::NondecimalBase::setBase(@_); }
+	sub convertBase { context::NondecimalBase::convert(@_); }
+}
+
+###########################################################################
+
+package context::NondecimalBase;
+
+# defines the base for the context.
+our $base = 10;
+#
+#  The standard digits, pre-built so it doesn't have to be done each time the conversion is called
+#
+our $digits16 = [ '0' .. '9', 'A' .. 'F' ];
+our $digit16  = { map { ($digits16->[$_], $_) } (0 .. scalar(@$digits16) - 1) };
+
+use Data::Dumper;
+#
+#  Initialize the contexts and make the creator function.
+#
+sub Init {
+	my $context = $main::context{NondecimalBase} = Parser::Context->getCopy("Numeric");
+	$context->{name} = 'NondecimalBase';
+	$context->{parser}{Number} = 'context::NondecimalBase::Number';
+	# $context->{value}{Real} = 'context::Hex::Hex';
+	$context->{pattern}{number} = '[0-9A-F]+';
+	$context->functions->disable('All');
+	$context->operators->remove('^');
+	$context->parens->remove('|');
+	$context->constants->clear();
+	# $context->operators->add(
+	#   '&'  => {precedence => .6, associativity => 'left', type => 'bin', string => ' & ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] & $_[1]}},
+	#   '|'  => {precedence => .5, associativity => 'left', type => 'bin', string => ' | ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] | $_[1]}},
+	#   '^'  => {precedence => .5, associativity => 'left', type => 'bin', string => ' ^ ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] ^ $_[1]}},
+	#   '>>' => {precedence => .4, associativity => 'left', type => 'bin', string => ' >> ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] >> $_[1]}},
+	#   '<<' => {precedence => .4, associativity => 'left', type => 'bin', string => ' << ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] << $_[1]}},
+	#   '~'  => {precedence => 6, associativity => 'left', type => 'unary', string => '~',
+	#            class => 'context::Hex::UOP::not'},
+	# );
+	# $context->{precedence}{Hex} = $context->{precedence}{special};
```

Comment:
You will need to use

``` perl
$context->{precedence}{NondecimalBase} = $context->{precedence}{special};
```

in order to make sure that operations involving your new `context::NondecimalBase::Real` will be handled by your code (rather than bu the other MathObject involved in the operation).